src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/NewAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,72 @@
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.WindowsAzure.Commands.Utilities.Common;
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Creates a new deployment.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRmDeployment", SupportsShouldProcess = true,
+        DefaultParameterSetName = ParameterlessTemplateFileParameterSetName), OutputType(typeof(PSDeployment))]
+    public class NewAzureDeploymentCmdlet : ResourceWithParameterCmdletBase, IDynamicParameters
+    {
+        [Alias("DeploymentName")]
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The name of the deployment it's going to create. Only valid when a template is used. When a template is used, if the user doesn't specify a deployment name, use the current time, like \"20131223140835\".")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "The location to store deployment data.")]
+        [LocationCompleter("Microsoft.Resources/resourceGroups")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The deployment debug log level.")]
+        [ValidateSet("RequestContent", "ResponseContent", "All", "None", IgnoreCase = true)]
+        public string DeploymentDebugLogLevel { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Do not ask for confirmation.")]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            var parameters = new PSDeploymentCmdletParameters()
+            {
+                Location = Location,
+                DeploymentName = Name,
+                DeploymentMode = DeploymentMode.Incremental,
+                TemplateFile = TemplateUri ?? this.TryResolvePath(TemplateFile),
+                TemplateParameterObject = GetTemplateParameterObject(TemplateParameterObject),
+                ParameterUri = TemplateParameterUri,
+                DeploymentDebugLogLevel = GetDeploymentDebugLogLevel(DeploymentDebugLogLevel)
+            };
+
+            if (!string.IsNullOrEmpty(parameters.DeploymentDebugLogLevel))
+            {
+                WriteWarning(ProjectResources.WarnOnDeploymentDebugSetting);
+            }
+            WriteObject(ResourceManagerSdkClient.ExecuteDeploymentAtSubscriptionScope(parameters));
```

Comment:
@Tiano2017 it doesn't look like this call isn't wrapped in a `ShouldProcess` call, so if the user provided `-WhatIf` to see what the cmdlet would do, it would actually execute the command

Reply:
do you have an example of how that works?

Reply:
@Tiano2017 you do it below for your [`Save-AzureRmDeployment`](https://github.com/Azure/azure-powershell/pull/6516/files#diff-f52bc2a8651b755def6e557531f08ed9R74) cmdlet 😀

Reply:
sorry what's the suggestion here?

---
In reply to: [207689032](https://github.com/Azure/azure-powershell/pull/6516#discussion_r207689032) [](ancestors = 207689032)

Reply:
addressed. 

---
In reply to: [205823255](https://github.com/Azure/azure-powershell/pull/6516#discussion_r205823255) [](ancestors = 205823255)